### DISCUSSION TITLE
Proposed Solution for Next.js Issue

### DISCUSSION THREAD

**Discussion by cholazzzb** [nick-keller/react-datasheet-grid#110] [General]
Hi, I just read this closed issue [Remove CSS import in order to support Next.js](https://github.com/Equify/react-datasheet-grid/issues/99). I come up with the idea that this issue can be solved with CSS-in-JS, for example using the simplest one is inline styles or style={{color: 'red'}} in the component. However, I see that the library makes use of CSS variables. Maybe the CSS variable can be moved inside the component with js as well? What are your thoughts?

**Comment by nick-keller** [comment]
One of the advantage of CSS variables is [letting the user customize them](https://react-datasheet-grid.netlify.app/docs/styling) at which ever level it desire.
Same for CSS classes, very easy to override a particular class.

I think the only solution would be to introduce a breaking change and force the users to import the style themself.

**Reply by LoncaSorin** [reply]
Couldn't we follow the example from here https://www.npmjs.com/package/react-datasheet-grid-nextjs? This works, but it doesn't have the latest updates

**Reply by nick-keller** [reply]
Hello @LoncaSorin , this will be live in less than 24h, thx for your support!

**Reply by LoncaSorin** [reply]
You're welcome, the library is great and I really wanted to use it. Keep up the good work!

**Comment by nick-keller** [comment]
Alright this is now live ;)

**Reply by LoncaSorin** [reply]
For me, it's still not working
![image](https://user-images.githubusercontent.com/18307908/151155915-a370ced7-22de-4f1e-ab5c-f44dedf54ed2.png)
I've seen that in the react-datasheet-grid-nextjs is used required instead of import, might be this the reason?
The error is coming from node_modules/react-datasheet-grid/dist/index.js but might be in other files as well

**Reply by nick-keller** [reply]
That has not changed since v3, I guess I need to compile imports into requires, if you have any insight that'd help a lot.

**Reply by LoncaSorin** [reply]
Sorry, but I don't. I will ask some of my colleagues and if I get an answer, I will come back and tell you.

**Reply by LoncaSorin** [reply]
@nick-keller sorry, I don't want to put pressure on you or something because you were great and answered so fast, but tomorrow I will need to present the library to the team in order to decide with which one should we go and it would really help if I could tell them how much time do you think it would take to compile the imports into requires? Or if you consider doing that
In any case, what you've built here is amazing and I would love to use it

**Reply by nick-keller** [reply]
@LoncaSorin we should be good now ;)

**Reply by LoncaSorin** [reply]
Omg, it's working, thank you, you're amazing!
I would send you a beer, I swear :))
